### PR TITLE
[#94479454] Added more flexibility to constructor

### PR DESCRIPTION
**Pull request by looser990**

### CODE REVIEW

File: java/messages/src/main/java/com/nedap/retail/messages/epcis/v1_1/EpcisEventContainer.java [hunk 1]
```diff
@@ -17,12 +18,11 @@
...
-    public EpcisEventContainer(final List<EpcisEvent> events) {
-        this.events = events;
+    public EpcisEventContainer(final List<? extends EpcisEvent> events) {
+        this.events = new ArrayList<>(events);
     }
 
     public EpcisEventContainer(final EpcisEvent epcisEvent) {
-        events = new ArrayList<>();
-        events.add(epcisEvent);
+        events = new ArrayList<>(Arrays.asList(epcisEvent));
```
Comment (stanislavgrujic) [inline java/messages/src/main/java/com/nedap/retail/messages/epcis/v1_1/EpcisEventContainer.java]: Arrays.asList() already returns a new ArrayList, there is no need to create another one.

File: java/messages/src/main/java/com/nedap/retail/messages/epcis/v1_1/EpcisEventContainer.java [hunk 2]
```diff
@@ -17,12 +18,11 @@
     public EpcisEventContainer() {
     }
 
-    public EpcisEventContainer(final List<EpcisEvent> events) {
-        this.events = events;
+    public EpcisEventContainer(final List<? extends EpcisEvent> events) {
+        this.events = new ArrayList<>(events);
```
Comment (jasperkremer) [inline java/messages/src/main/java/com/nedap/retail/messages/epcis/v1_1/EpcisEventContainer.java]: Vague comment: _"Added more flexibility to constructor"_.
What is the reason behind this change?

Comment (looser990) [inline java/messages/src/main/java/com/nedap/retail/messages/epcis/v1_1/EpcisEventContainer.java]: So we can create EpcisEventContainer with `new EpcisEventContainer(events)` where `events` are any of the possible type for EpcisEvent.

Comment (jasperkremer) [inline java/messages/src/main/java/com/nedap/retail/messages/epcis/v1_1/EpcisEventContainer.java]: It seems like an improper usage of using generics in this case.
See also http://stackoverflow.com/questions/1292109/generics-get-and-put-rule

Why not do something like this:
```
  public class EventContainer {

        public List<EpcisEvent> events;

        public EventContainer() {
            events = new ArrayList<>();
        }

        public EventContainer(List<EpcisEvent> events) {
            this.events = events;
        }

        public EventContainer(EpcisEvent epcisEvent) {
            this(Arrays.asList(epcisEvent));
        }

        public EventContainer add(List<ObjectEvent> events) {
            events.addAll(events);
        }
    }

    private void foo() {
        List<ObjectEvent> list = new ArrayList<>();
        EventContainer eventsContainer = new EventContainer().add(list);
    }
```
